src/darwin/Framework/CHIP/CHIPDeviceController+XPC.m

```diff
@@ -111,6 +115,54 @@ + (CHIPDeviceController *)sharedControllerWithId:(id<NSCopying> _Nullable)contro
     return result;
 }
 
++ (NSDictionary<NSString *, id> * _Nullable)encodeXPCReadParams:(CHIPReadParams *)params
+{
+    if (!params) {
+        return nil;
+    }
+    NSMutableDictionary<NSString *, id> * result = [NSMutableDictionary dictionary];
+    if (params.fabricFiltered) {
+        result[kFabricFilteredKey] = params.fabricFiltered;
+    }
+    return result;
+}
+
++ (CHIPReadParams * _Nullable)decodeXPCReadParams:(NSDictionary<NSString *, id> * _Nullable)params
+{
+    if (!params) {
+        return nil;
+    }
+    CHIPReadParams * result = [[CHIPReadParams alloc] init];
+    result.fabricFiltered = params[kFabricFilteredKey];
+    return result;
+}
+
++ (NSDictionary<NSString *, id> * _Nullable)encodeXPCSubscribeParams:(CHIPSubscribeParams *)params
+{
+    if (!params) {
+        return nil;
+    }
+    NSMutableDictionary<NSString *, id> * result = [NSMutableDictionary dictionary];
+    if (params.fabricFiltered) {
+        result[kFabricFilteredKey] = params.fabricFiltered;
+    }
```

Comment:
Would it make sense here to share code by calling `encodeXPCReadParams`?

Also, this needs to be updated to handle `autoResubscribe`.